cli/render.go

```diff
@@ -15,9 +24,134 @@ import (
 type RenderCommand struct {
 	*baseCommand
 	packConfig *cache.PackConfig
-	// renderOutputTemplate is a boolean flag to control whether the outpu
+	// renderOutputTemplate is a boolean flag to control whether the output
 	// template is rendered.
 	renderOutputTemplate bool
+	// renderToDir is the path to write rendered job files to in addition to
+	// standard output.
+	renderToDir string
+}
+
+type Render struct {
+	Name    string
+	Content string
+}
+
+func (r Render) toTerminal(c *RenderCommand) {
+	c.ui.Output(r.Name+":", terminal.WithStyle(terminal.BoldStyle))
+	c.ui.Output("")
+	c.ui.Output(r.Content)
+}
+
+func (r Render) toFile(c *RenderCommand, ec *errors.UIErrorContext) error {
+	renderToDir := path.Clean(c.renderToDir)
+	err := validateOutDir(renderToDir)
+	if err != nil {
+		ec.Add("Destination Dir: ", renderToDir)
+		return err
+	}
+
+	filePath, fileName := path.Split(r.Name)
+	outDir := path.Join(renderToDir, filePath)
+	outFile := path.Join(outDir, fileName)
+
+	maybeCreateDestinationDir(outDir)
+
+	var overwrite bool
+
+	if !c.autoApproved && c.ui.Interactive() {
+		overwrite, err = confirmOverwrite(c)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = writeFile(outFile, r.Content, c.autoApproved || overwrite)
+	if err != nil {
+		ec.Add("Destination File: ", outFile)
+		return err
+	}
+
+	return nil
+}
+
+func confirmOverwrite(c *RenderCommand) (bool, error) {
+	for {
+		overwrite, err := c.ui.Input(&terminal.Input{
+			Prompt: "Output file exists, overwrite? [y/n] ",
+			Style:  terminal.WarningBoldStyle,
+		})
+		if err != nil {
+			return false, err
+		}
+		overwrite = strings.ToLower(overwrite)
+		if overwrite == "y" || overwrite == "n" {
+			return overwrite == "y", nil
+		}
+	}
+
+}
+
+func validateOutDir(path string) error {
+	if path == "" {
+		return nil
+	}
+	info, err := os.Stat(path)
+
+	if err != nil {
+		if stdErrors.Is(err, fs.ErrNotExist) {
+			return nil
+		}
+
+		return fmt.Errorf("unexpected error testing --to-dir path: %w", err)
+	}
+
+	if !info.IsDir() {
+		return stdErrors.New("--to-dir must be a directory")
+	}
+
+	return nil
+}
+
+func maybeCreateDestinationDir(path string) error {
+	_, err := os.Stat(path)
+
+	// If the directory doesn't exist, create it.
+	if stdErrors.Is(err, fs.ErrNotExist) {
+		err := os.MkdirAll(path, 0755)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func writeFile(path string, content string, overwrite bool) error {
+	// Check to see if the file already exists and validate against the value
+	// of overwrite.
+
+	_, err := os.Stat(path)
+
+	if err == nil && !overwrite {
+		return fmt.Errorf("destination file exists and overwrite is unset")
+	}
+
+	err = ioutil.WriteFile(path, []byte(content), 0644)
```

Comment:
`ioutil` is deprecated and goes away as of go 1.17 I believe. Should we update this now to use `os` functions to be future proof?